packages/devextreme/js/__internal/ui/m_tag_box.ts

```diff
@@ -831,10 +841,23 @@ const TagBox = (SelectBox as any).inherit({
       const currentItem = filteredValues[isValueExprSpecified ? JSON.stringify(value) : value];
 
       if (isValueExprSpecified && !isDefined(currentItem)) {
-        loadItemPromises.push(this._loadItem(value, cache).always((item) => {
-          const newItem = this._createTagData(item, value);
-          items.splice(index, 0, newItem as never);
-        }) as never);
+        if (!this._dataSource) {
+          return;
+        }
+
+        loadItemPromises.push(
+          this._loadItem(value, cache)
+            .done((item) => {
+              const newItem = this._createTagData(item, value);
+              items.splice(index, 0, newItem as never);
```

Comment:
This cast for the newItem type looks like a hack. Please add the correct types. 
If that's difficult for you, at least use a // @ts-expect-error comment instead in all places.

Reply:
add comment here and in other places, i don't want to think on types updates in scope of this task which is arguable enough itself